packages/tfm-ror52-rubygem-thor/tfm-ror52-rubygem-thor.spec

```diff
@@ -0,0 +1,95 @@
+# Generated from thor-0.20.0.gem by gem2rpm -*- rpm-spec -*-
+%{?scl:%scl_package rubygem-%{gem_name}}
+%{!?scl:%global pkg_name %{name}}
+
+%global gem_name thor
+
+Name:    %{?scl_prefix}rubygem-%{gem_name}
+Version: 0.20.0
+Release: 1%{?dist}
+Summary: Thor is a toolkit for building powerful command-line interfaces
+Group:   Development/Languages
+License: MIT
+URL:     http://whatisthor.com/
+Source0: https://rubygems.org/gems/%{gem_name}-%{version}.gem
+
+Requires: %{?scl_prefix_ruby}ruby(release)
+Requires: %{?scl_prefix_ruby}ruby(rubygems)
+BuildRequires: %{?scl_prefix}build
+BuildRequires: %{?scl_prefix_ruby}ruby(release)
+BuildRequires: %{?scl_prefix_ruby}rubygems-devel >= 1.3.5
+BuildRequires: %{?scl_prefix_ruby}ruby >= 1.8.7
+BuildArch: noarch
+Provides: %{?scl_prefix}rubygem(%{gem_name}) = %{version}
+
+%{?scl:Requires: %{?scl_prefix}runtime}
+
+%description
+Thor is a toolkit for building powerful command-line interfaces.
+
+
+%package doc
+Summary:   Documentation for %{name}
+Group:     Documentation
+Requires:  %{?scl_prefix}%{pkg_name} = %{version}-%{release}
+BuildArch: noarch
+
+%description doc
+Documentation for %{name}.
+
+%prep
+%{?scl:scl enable %{scl} - << \EOF}
+gem unpack %{SOURCE0}
+%{?scl:EOF}
+
+%setup -q -D -T -n  %{gem_name}-%{version}
+
+%{?scl:scl enable %{scl} - << \EOF}
+gem spec %{SOURCE0} -l --ruby > %{gem_name}.gemspec
+%{?scl:EOF}
+
+%build
+# Create the gem as gem install only works on a gem file
+%{?scl:scl enable %{scl} - << \EOF}
+gem build %{gem_name}.gemspec
+%{?scl:EOF}
+
+# %%gem_install compiles any C extensions and installs the gem into ./%%gem_dir
+# by default, so that we can move it into the buildroot in %%install
+%{?scl:scl enable %{scl} - << \EOF}
+# Work around incorrect bindir set during gem install
+mkdir -p ./opt/rh/rh-ruby25/root/usr/bin
```

Comment:
I'm ok with this workaround pending further investigation.

Reply:
I'd prefer if we could at least use `%{scl_ruby}` instead of `rh-ruby25` but I'm pretty sure you can use `%{_scl_root}/usr/bin` for this. Reading the macros it should even define `%{_bindir}` as `/opt/rh/rh-ruby25/root/usr/bin`

Reply:
This won't work because we are in the context of tfm-ror52 SCL and thus every macro is aligned to it's pathing not the Ruby SCLs pathing. Thus, hard coding is the only thing that works outside of defining a global at the top of the file

Reply:
At least `scl_ruby` is still `rh-ruby25`. Our SCL does define `scl_ror` as `tfm-ror52`. With all the build deps installed:
```
[root@d329c535c3db /]# rpm --eval '%{scl_ruby}'
rh-ruby25
```

However, what I find weird is that the `PATH` in our SCL is odd:
```
[root@d329c535c3db /]# scl enable tfm-ror52 'echo $PATH'
/opt/rh/rh-ruby25/root/usr/local/bin:/opt/rh/rh-ruby25/root/usr/bin:/opt/theforeman/tfm-ror52/root/usr/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```

I would expect tfm-ror52 to be before the ruby one and bins from this gem end up in `/opt/theforeman/tfm-ror52/root/usr/bin`. Should this be in another order:
```
[root@d329c535c3db /]# cat /opt/theforeman/tfm-ror52/enable 
export PATH=/opt/theforeman/tfm-ror52/root/usr/bin${PATH:+:${PATH}}
export LD_LIBRARY_PATH=/opt/theforeman/tfm-ror52/root/usr/lib64${LD_LIBRARY_PATH:+:${LD_LIBRARY_PATH}}
export MANPATH=/opt/theforeman/tfm-ror52/root/usr/share/man:${MANPATH}
export PKG_CONFIG_PATH=/opt/theforeman/tfm-ror52/root/usr/lib64/pkgconfig${PKG_CONFIG_PATH:+:${PKG_CONFIG_PATH}}
export GEM_PATH=${GEM_PATH:=/opt/theforeman/tfm-ror52/root/usr/share/gems:`scl enable rh-ruby25 -- ruby -e "print Gem.path.join(':')"`}

. scl_source enable rh-ruby25
```

i.e. first source rh-ruby25 and then set variables. In `tfm-ror51` we do the same thing so that shouldn't explain the difference here though.